.github/workflows/poetry.yml

```diff
@@ -16,7 +16,7 @@ jobs:
       max-parallel: 9
       fail-fast: false
       matrix:
-        python-version: ["3.8", "3.9", "3.10", "3.11"]
+        python-version: ["3.11"]
```

Comment:
So... technically, since versions of our dependencies might be released only for specific versions of Python, is it worthwhile for us to keep all these checks?

Reply:
That would only work if we had a separate lock file for each version.

Reply:
I see... is that something we should consider doing?

Reply:
A) I have no idea how you’d manage that.
B) afaik the lock file should be the same for every version. I contains rules for which deps go to which versions of python.

Reply:
As a refresher for what this action does: 

- `poetry check` verifies the pyproject.toml has proper structure 
- `poetry lock --check` Verify that poetry.lock is consistent with pyproject.toml

This was mean to ensure that in the case when the `pyproject` got updated, a new lock would also get created.  The lock contains all of the versions possible based on the rules set in the `pyproject`.   So even if the `pyproject` had a dependency like `typing_extensions>=3.10.0.0; python_version < '3.10'`, there would still only be a single lock file.   

Technically speaking, we don't NEED to include the lock in the repo.  I did that so we could be in sync with our packages during development.  That way when we move between branches we could come back to the exact same dependencies.

Reply:
I think having the lock committed also allows us to verify the correct SHAs for the packages we have installed and developed against, which is a security feature!